src/python/CRABClient/Commands/remake.py

```diff
@@ -36,8 +37,15 @@ def remakecache(self,taskname):
                          'RequestName': taskname, 'voRole': '', 'Port': ''}, dumpfile, protocol=0)
             dumpfile.close()
             self.logger.info("%sSuccess%s: Finished remaking project directory %s" % (colors.GREEN, colors.NORMAL, requestarea))
-            return requestarea
 
+            caller = sys._getframe(1)
+            if not 'crab.py' in caller.f_code.co_filename:
+                msg = "ATTENTION return value for 'remake' has been changed to a dictionary"
+                msg += "\n format is {'commandStatus': 'SUCCESS' or 'FAILED',"
+                msg += "\n            'workDir': name of the work directory created}"
+                self.logger.warning(msg)
+            returnDict = {'commandStatus': 'SUCCESS', 'workDir': requestarea}
+            return returnDict
```

Comment:
No need of warning message if you do not support backward compatibility.

Reply:
I hope that a warning + a failure is more clear than just an obscure failure on some code which used to work. It surely makes no sense long term, but simply to help the transition. I do not think I have a way to catch if user code fails on my new return value or not.